docs_src/microservices/general/messagebus.md

```diff
@@ -0,0 +1,368 @@
+    The following `MessageQueue` configuration settings must be changed for all EdgeX Core/Support/Device Services to use NATS Jetstream
+
+    ```toml
+    [MessageQueue]
+    Type = "nats-jetstream"
+    Protocol = "tcp" 
+    Host = "localhost" # in docker this must be overriden to be the docker host name of the MQTT Broker
+    Port = 4222
+    AuthMode = "none"  # Currently in secure mode the NATS server is not secured
+      [MessageQueue.Optional]
+      ClientId ="<service-key>" # must be unique name of the service, thus the service key (core-data, etc) is used
+      # Connection information
+      Format =  "nats" # Use 'json' for backward compatability with services using MQTT
+      ConnectTimeout = "5" # Seconds
+      RetryOnFailedConnect = "true"
+      QueueGroup = "" 
+      Durable =  "" # Jetstream only
+      AutoProvision = "true" # Jetstream only
+      Deliver = "new" # Jetstream only
+    ```
+
+!!! example - "Example NATS Configurations changes for Application Services"
+    The following `Trigger.EdgexMessageBus` configuration settings must be changed for all EdgeX Application Services to use NATsJetstream
+
+    ```toml
+    [Trigger]
+    Type="edgex-messagebus"
+      [Trigger.EdgexMessageBus]
+      Type = "nats-jetstream"
+        [Trigger.EdgexMessageBus.SubscribeHost]
+        Protocol = "tcp"
+        Host = "localhost" # in docker this must be overriden to be the docker host name of the MQTT Broker
+        Port = 4222
+        [Trigger.EdgexMessageBus.PublishHost]
+        Protocol = "tcp"
+        Host = "localhost" # in docker this must be overriden to be the docker host name of the MQTT Broker
+        Port = 4222
+        [Trigger.EdgexMessageBus.Optional]
+        ClientId ="<service-key>" # must be unique name of the service, thus the service key (app-rule-engine, etc) is used
+        Format =  "nats" # Use 'json' for backward compatability with services using MQTT
+        ConnectTimeout = "5" # Seconds
+        RetryOnFailedConnect = "true"
+        QueueGroup = "" 
+        Durable =  "" # Jetstream only
+        AutoProvision = "true" # Jetstream only
+        Deliver = "new" # Jetstream only
+        authmode = "none"  # Currently in secure mode the NATS Server is not secured
+    ```
+
+#### Docker
+
+The EdgeX Compose Builder utility provides an option to easily generate a compose file with all the selected services re-configured for NATS using environment overrides. This is accomplished by using the `nats-bus` option. This option configures the services to use the NATS Jetstream implementation. See [Compose Builder README](https://github.com/edgexfoundry/edgex-compose/tree/{{latest_release_name}}/compose-builder) for details on all available options. If NATS Core is preferred, simply do a search and replace of `nats-jeststream` with `nats-core` in the generated compose file.
+
+!!! example - "Example Secure mode compose generation for NATS"
+
+    ```
+    make gen ds-virtual ds-rest nats-bus

```

Comment:
Love it!  Do we need to add empty properties to be overriden in the `optional` maps for any config files to support this?

Reply:
> Do we need to add empty properties to be overriden in the `optional` maps for any config files to support this?

Yes those will need to be added to all the TOML files. I will be adding Make targets to build with NATS and can include then then.